src/OrbitGl/PickingManager.h

```diff
@@ -51,41 +45,70 @@ enum class PickingType : uint32_t {
 // in the bits remaining after encoding the batcher id, and the
 // PickingType, so adding more batchers or types has to be carefully
 // considered.
-enum class BatcherId : uint32_t { kTimeGraph, kUi };
+enum class BatcherId { kTimeGraph, kUi, kCount };
 
 struct PickingId {
   static constexpr uint32_t kElementIDBitSize = 28;
   static constexpr uint32_t kPickingTypeBitSize = 3;
   static constexpr uint32_t kBatcherIDBitSize = 1;
 
-  static_assert(kElementIDBitSize + kPickingTypeBitSize + kBatcherIDBitSize == 32);
+  struct Layout {
+    uint32_t element_id : kElementIDBitSize;
+    uint32_t type : kPickingTypeBitSize;
+    uint32_t batcher_id : kBatcherIDBitSize;
+  };
+
+  static_assert(sizeof(Layout) == sizeof(uint32_t),
+                "Layout needs to have the size of a single uint32_t.");
+  static_assert(1 << kPickingTypeBitSize >= static_cast<int>(PickingType::kCount),
+                "PickingType cannot be encoded in the specified number of bits.");
+  static_assert(1 << kBatcherIDBitSize >= static_cast<int>(BatcherId::kCount),
+                "BatcherId cannot be encoded in the specified number of bits.");
 
   [[nodiscard]] inline static PickingId Create(PickingType type, uint32_t element_id,
                                                BatcherId batcher_id = BatcherId::kTimeGraph) {
     CHECK(element_id >> kElementIDBitSize == 0);
```

Comment:
add CHECK for batcher_id?

Reply:
Done and also added check for picking type.